fixture-monkey/src/main/java/com/navercorp/fixturemonkey/expression/StrictModeNextNodePredicateContainer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Fixture Monkey
+ *
+ * Copyright (c) 2021-present NAVER Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.fixturemonkey.expression;
+
+import java.lang.reflect.Field;
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import com.navercorp.fixturemonkey.api.property.FieldProperty;
+import com.navercorp.fixturemonkey.api.property.PropertyNameResolver;
+import com.navercorp.fixturemonkey.api.type.Types;
+import com.navercorp.fixturemonkey.tree.NextNodePredicate;
+import com.navercorp.fixturemonkey.tree.PropertyNameNodePredicate;
+
+public final class StrictModeNextNodePredicateContainer extends AbstractList<NextNodePredicate> {
+	private final List<NextNodePredicate> delegate;
+
+	public StrictModeNextNodePredicateContainer(
+		List<NextNodePredicate> delegate, Class<?> rootClass, PropertyNameResolver propertyNameResolver
+	) {
+		this.delegate = delegate;
+		if (!isValidFieldPath(rootClass, delegate, propertyNameResolver)) {
+			throw new IllegalArgumentException("No matching results for given container expression.");
+		}
+	}
+
+	@Override
+	public NextNodePredicate get(int index) {
+		return delegate.get(index);
+	}
+
+	@Override
+	public int size() {
+		return delegate.size();
+	}
+
+	private boolean isValidFieldPath(
```

Comment:
I think there is a limitation in that it cannot check the element.

For example, it would not throw an exception in the following case:

```
**String actual = sut.giveMeBuilder(new TypeReference<List<List<SimpleObject>>>() {
			})
			.size("$", 1)
			.size("$[0]", 1)
			.set("$[0][1].str", "expected")
			.sample()
			.get(0)
			.get(0)
			.getStr();
```

Nevertheless, it is a useful feature, so adding a comment to describe the limitation would be better.

Reply:
Sorry, I've misunderstood. It is worked with the existing `MonkeyExpresionFactory`.

I mean the below case.

```
FixtureMonkey sut = FixtureMonkey.builder().useExpressionStrictMode().build();

		List<SimpleObject> actual = sut.giveMeBuilder(new TypeReference<List<List<SimpleObject>>>() {
			})
			.size("$", 1)
			.size("$[1]", 1)
			.sample()
			.get(0);
```